subprojects/base-services/src/main/java/org/gradle/internal/progress/BuildOperationDescriptor.java

```diff
@@ -23,14 +23,14 @@
  * Meta-data about a build operation.
  */
 public class BuildOperationDescriptor {
-    private final Object id;
-    private final Object parentId;
+    private final OperationIdentifier id;
```

Comment:
We don't necessarily need to strongly type the build operation ids to make logging work well. I would do that as a separate move.